### PR TITLE
used docker golang to build prombench

### LINKED ISSUE #145: Let's use Docker to build the binaries.
instead or relying that the local dev env has golang with modules support installed let's just use the minimal [golang docker image](https://hub.docker.com/_/golang/)

https://github.com/prometheus/prombench/blob/f13a14db5a1d3bee58999ff789cc58ff593ff22a/Makefile#L31-L33

### PR DESCRIPTION
**Pull request by geekodour**
Cont. #146 
Closes #145 

Signed-off-by: geekodour <hrishikeshbman@gmail.com>

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
actually can you check if we can use promu for this?

**Comment by krasi-georgiev** [comment]
hm doesn't look like promu has a docker image.

**Comment by geekodour** [comment]
@krasi-georgiev I am not feeling good about having to use `sudo` to change ownership of the executable but if I run the golang docker image as the running user then it shows something like `cannot use internal cache, required for gomodules` which is reasonable as internal cache only the container root user has access to.

Is having the `sudo` command there fine?

**Comment by krasi-georgiev** [comment]
Can you first check if promu has a docker image and if yes I guess the problem is already solved there. Ask in the IRC -dev channel. I think @SuperQ or @simonpasquier can answer that question.

**Comment by krasi-georgiev** [comment]
after seeing Ben's comment on IRC it sounds it would be easiest to  stick to this simple approach of using the docker golang image directly.


>  I am not feeling good about having to use sudo to change ownership of the executable but if I run the golang docker image as the running user then it shows something like cannot use internal cache, required for gomodules which is reasonable as internal cache only the container root user has access to.

when running docker you can map users/groups from the host so this should avoid the need to use sudo. Sudo is another dependency and I like to minimise dependencies.

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ GOLANG_IMG = golang:1.11
 PROMBENCH_DIR = /go/src/github.com/prometheus/prombench
 USERID = $(shell id -u ${USER})
 USERGROUP = $(shell id -g ${USER})
-DOCKER_CMD = docker run --rm \
+DOCKER_CMD = docker run --rm\
```
Comment (krasi-georgiev) [inline Makefile]: instead of `@$(DOCKER_CMD) chown ${USERID}:${USERGROUP} `
you should be able to do `@$(DOCKER_CMD) = docker run -u $USERID:$USERGROUP`

Comment (geekodour) [reply Makefile]: I tried doing that earlier, but it gives me the following error:
```
go: disabling cache (/.cache/go-build) due to initialization failure: mkdir /.cache: permission denied
go: cannot use modules with build cache disabled
```

Comment (krasi-georgiev) [reply Makefile]: ok go it , lets leave as is.

Comment (krasi-georgiev) [reply Makefile]: maybe try setting GOCACHE or HOME envs

https://github.com/golang/go/issues/26280

Comment (krasi-georgiev) [reply Makefile]: https://golang.org/pkg/cmd/go/internal/help/

Comment (krasi-georgiev) [reply Makefile]: you can set GOCACHE to /tmp which is writable by everyone

Comment (geekodour) [reply Makefile]: Yes i just did that, it's working now.

**Comment by krasi-georgiev** [comment]
LGTM
ping @sipian

**Review by sipian — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -1,39 +1,52 @@
 PROMBENCH_CMD        = ./prombench
 DOCKER_TAG = docker.io/prombench/prombench:2.0.0
+GOLANG_IMG = golang:1.11
```
Comment (krasi-georgiev) [inline Makefile]: why not 1.12? I just tested it and it seems to be working fine.

**Comment by krasi-georgiev** [comment]
also I just remembered that it would be good to mention in https://github.com/prometheus/prombench/blob/master/cmd/prombench/README.md that can use `make build` to build it.

**Comment by krasi-georgiev** [comment]
Thanks!

**Pull request merged**